pkg/scheduler/plugins/deviceshare/device_cache.go

```diff
@@ -382,15 +382,20 @@ func newNodeDeviceCache() *nodeDeviceCache {
 	}
 }
 
-func (n *nodeDeviceCache) getNodeDevice(nodeName string) *nodeDevice {
-	n.lock.RLock()
-	defer n.lock.RUnlock()
+func (n *nodeDeviceCache) getNodeDevice(nodeName string, needInit bool) *nodeDevice {
+	n.lock.Lock()
+	defer n.lock.Unlock()
+
+	// getNodeDevice will call createNodeDevice(nodeName) if needInit is true and nodeDeviceInfos[nodeName] is nil
+	if n.nodeDeviceInfos[nodeName] == nil && needInit {
+		klog.V(5).Infof("node device cache not found, nodeName: %v, createNodeDevice", nodeName)
+		return n.createNodeDevice(nodeName)
+	}
+
 	return n.nodeDeviceInfos[nodeName]
 }
 
 func (n *nodeDeviceCache) createNodeDevice(nodeName string) *nodeDevice {
```

Comment:
It is recommended to remove this function and merge with getNodeDevice.

Reply:
sure, no problem.